hibernate-core/src/main/java/org/hibernate/type/Type.java

```diff
@@ -108,7 +108,7 @@
 	 *
 	 * @throws MappingException Generally indicates an issue accessing the passed mapping object.
 	 */
-	int[] sqlTypes(Mapping mapping) throws MappingException;
+	int[] getSqlTypeCodes(Mapping mapping) throws MappingException;
```

Comment:
As much as I like this change, I think this is a backwards compatibility concern. Not sure if that is something we are ok with, but maybe revert the renaming here?

Reply:
Depends what @sebersole thinks.

Reply:
I do think it would be nice to minimize changes as much as we can.

Here though, I think we should just make this change.  This contract is changing already wrt read-from-jdbc , so implementors and consumers are going to have to change.  If we do the whole deprecation paradigm, we are just making them make changes again later if/when we remove the deprecated form)